---
title: Clase Exception y propiedades
ms.date: 03/30/2017
helpviewer_keywords:
- exceptions, Exception class
- Exception class
ms.assetid: e2e1f8c4-e7b4-467d-9a66-13c90861221d
ms.openlocfilehash: 459dc15f136c98d4b889a420dbd863b16aca747c
ms.sourcegitcommit: 965a5af7918acb0a3fd3baf342e15d511ef75188
ms.translationtype: HT
ms.contentlocale: es-ES
ms.lasthandoff: 11/18/2020
ms.locfileid: "94828210"
---
# <a name="exception-class-and-properties"></a><span data-ttu-id="8b0ff-102">Clase Exception y propiedades</span><span class="sxs-lookup"><span data-stu-id="8b0ff-102">Exception class and properties</span></span>

<span data-ttu-id="8b0ff-103">La clase <xref:System.Exception> es la clase base de la que heredan las excepciones.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-103">The <xref:System.Exception> class is the base class from which exceptions inherit.</span></span> <span data-ttu-id="8b0ff-104">Por ejemplo, la jerarquía de clases de <xref:System.InvalidCastException> es como sigue:</span><span class="sxs-lookup"><span data-stu-id="8b0ff-104">For example, the <xref:System.InvalidCastException> class hierarchy is as follows:</span></span>

<xref:System.Object>\
&nbsp;&nbsp;<xref:System.Exception>\
&nbsp;&nbsp;&nbsp;&nbsp;<xref:System.SystemException>\
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<xref:System.InvalidCastException>

<span data-ttu-id="8b0ff-105">La clase <xref:System.Exception> tiene las siguientes propiedades que facilitan la comprensión de una excepción.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-105">The <xref:System.Exception> class has the following properties that help make understanding an exception easier.</span></span>

| <span data-ttu-id="8b0ff-106">Nombre de la propiedad</span><span class="sxs-lookup"><span data-stu-id="8b0ff-106">Property Name</span></span> | <span data-ttu-id="8b0ff-107">Description</span><span class="sxs-lookup"><span data-stu-id="8b0ff-107">Description</span></span> |
| ------------- | ----------- |
| <xref:System.Exception.Data> | <span data-ttu-id="8b0ff-108">Un <xref:System.Collections.IDictionary> que contiene datos arbitrarios en pares de clave y valor.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-108">An <xref:System.Collections.IDictionary> that holds arbitrary data in key-value pairs.</span></span> |
| <xref:System.Exception.HelpLink> | <span data-ttu-id="8b0ff-109">Puede contener una dirección URL (o URN) a un archivo de ayuda que proporciona amplia información sobre la causa de una excepción.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-109">Can hold a URL (or URN) to a help file that provides extensive information about the cause of an exception.</span></span> |
| <xref:System.Exception.InnerException> | <span data-ttu-id="8b0ff-110">Esta propiedad puede usarse para crear y mantener una serie de excepciones durante el control de excepciones.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-110">This property can be used to create and preserve a series of exceptions during exception handling.</span></span> <span data-ttu-id="8b0ff-111">Puede usarla para crear una nueva excepción que contiene excepciones detectadas con anterioridad.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-111">You can use it to create a new exception that contains previously caught exceptions.</span></span> <span data-ttu-id="8b0ff-112">La segunda excepción en la propiedad <xref:System.Exception.InnerException> puede capturar la excepción original, lo que permite que el código que controla la segunda excepción examine la información adicional.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-112">The original exception can be captured by the second exception in the <xref:System.Exception.InnerException> property, allowing code that handles the second exception to examine the additional information.</span></span> <span data-ttu-id="8b0ff-113">Por ejemplo, supongamos que tiene un método que recibe un argumento que contiene un formato incorrecto.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-113">For example, suppose you have a method that receives an argument that's improperly formatted.</span></span>  <span data-ttu-id="8b0ff-114">El código intenta leer el argumento, pero se inicia una excepción.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-114">The code tries to read the argument, but an exception is thrown.</span></span> <span data-ttu-id="8b0ff-115">El método detecta la excepción e inicia <xref:System.FormatException>.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-115">The method catches the exception and throws a <xref:System.FormatException>.</span></span> <span data-ttu-id="8b0ff-116">Para mejorar la capacidad del autor de llamada a la hora de determinar por qué se inicia una excepción, a veces, para un método es recomendable detectar una excepción iniciada por una rutina del asistente y, después, iniciar una excepción más indicativa del error que se ha producido.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-116">To improve the caller's ability to determine the reason an exception is thrown, it is sometimes desirable for a method to catch an exception thrown by a helper routine and then throw an exception more indicative of the error that has occurred.</span></span> <span data-ttu-id="8b0ff-117">Puede crear una excepción nueva y más significativa, donde se puede establecer la referencia a la excepción interna en la excepción original.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-117">A new and more meaningful exception can be created, where the inner exception reference can be set to the original exception.</span></span> <span data-ttu-id="8b0ff-118">Después, se puede iniciar esta excepción más significativa para el autor de llamada.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-118">This more meaningful exception can then be thrown to the caller.</span></span> <span data-ttu-id="8b0ff-119">Tenga en cuenta que con esta funcionalidad, puede crear una serie de excepciones vinculadas que termina con la excepción que se inicia en primer lugar.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-119">Note that with this functionality, you can create a series of linked exceptions that ends with the exception that was thrown first.</span></span> |
| <xref:System.Exception.Message> | <span data-ttu-id="8b0ff-120">Proporciona detalles sobre la causa de una excepción.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-120">Provides details about the cause of an exception.</span></span>
| <xref:System.Exception.Source> | <span data-ttu-id="8b0ff-121">Devuelve o establece el nombre de la aplicación o del objeto que generó el error.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-121">Gets or sets the name of the application or the object that causes the error.</span></span> |
| <xref:System.Exception.StackTrace>| <span data-ttu-id="8b0ff-122">Contiene un seguimiento de la pila que puede usarse para determinar dónde se produjo un error.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-122">Contains a stack trace that can be used to determine where an error occurred.</span></span> <span data-ttu-id="8b0ff-123">El seguimiento de la pila contiene el nombre del archivo de código fuente y el número de línea del programa si está disponible la información de depuración.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-123">The stack trace includes the source file name and program line number if debugging information is available.</span></span> |

<span data-ttu-id="8b0ff-124">La mayoría de las clases que heredan de <xref:System.Exception> no implementan miembros adicionales ni proporcionan funciones adicionales; simplemente heredan de <xref:System.Exception>.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-124">Most of the classes that inherit from <xref:System.Exception> do not implement additional members or provide additional functionality; they simply inherit from <xref:System.Exception>.</span></span> <span data-ttu-id="8b0ff-125">Por tanto, se puede encontrar la información más importante para una excepción en la jerarquía de clases de excepción, el nombre de la excepción y la información contenida en la excepción.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-125">Therefore, the most important information for an exception can be found in the hierarchy of exception classes, the exception name, and the information contained in the exception.</span></span>

<span data-ttu-id="8b0ff-126">Se recomienda iniciar y detectar solo los objetos que se derivan de <xref:System.Exception>, pero puede iniciar cualquier objeto que se deriva de la clase <xref:System.Object> como una excepción.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-126">We recommend that you throw and catch only objects that derive from <xref:System.Exception>, but you can throw any object that derives from the <xref:System.Object> class as an exception.</span></span> <span data-ttu-id="8b0ff-127">Tenga en cuenta que no todos los lenguajes admiten iniciar y detectar objetos que no se derivan de <xref:System.Exception>.</span><span class="sxs-lookup"><span data-stu-id="8b0ff-127">Note that not all languages support throwing and catching objects that do not derive from <xref:System.Exception>.</span></span>
  
## <a name="see-also"></a><span data-ttu-id="8b0ff-128">Vea también</span><span class="sxs-lookup"><span data-stu-id="8b0ff-128">See also</span></span>

- [<span data-ttu-id="8b0ff-129">Excepciones</span><span class="sxs-lookup"><span data-stu-id="8b0ff-129">Exceptions</span></span>](index.md)
